### PR TITLE
[BE][c10d] Add a warning messages in the comment about cuda hang

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130844

Add comments to warn users potential hang for the cuda event query in NCCLPG.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by wconstab — APPROVED** [review]

**Comment by fduwjj** [comment]
Fixed lint error and grammar errors.

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 1]
```diff
@@ -568,6 +568,11 @@ bool ProcessGroupNCCL::WorkNCCL::startedGPUExecutionInternal() const {
 
 bool ProcessGroupNCCL::WorkNCCL::finishedGPUExecutionInternal() const {
   // Checking the work's corresponding CUDA event's status
+  // It calls `cudaEventQuery` eventually. Although this seems to be a
+  // non-blocking call, but we did notice hangs in the past. It can
```
Comment (awgu) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:572]: nit: 
```suggestion
  // non-blocking call, we noticed hangs in the past. It can
```

Comment (fduwjj) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp]: Opps, I forgot, lol

**Pull request closed without merging**